src/boost_histogram/_internal/hist.py

```diff
@@ -489,6 +505,11 @@ def fill(
             available threads (usually two per core).
         """
 
+        if self._hist._storage_type in {
+            _core.storage.mean,
+        }:
```

Comment:
```suggestion
        if self._hist._storage_type in {_core.storage.mean}:
```

Reply:
I guess weighted_mean has different input

Reply:
What was the problem?

Reply:
We expect 1D array for input in mean storage. However, I saw simple numerical inputs for weighted_mean storage in examples so we can't apply the mean_storage_sample_check for weighted_mean too!